crypto/asn1/a_time.c

```diff
@@ -489,7 +489,7 @@ int ASN1_TIME_print(BIO *bp, const ASN1_TIME *tm)
         if (tm->length > 15 && v[14] == '.') {
             f = &v[14];
             f_len = 1;
-            while (14 + f_len < l && isdigit(f[f_len]))
+            while (14 + f_len < l && ossl_isasciidigit(f[f_len]))
```

Comment:
The ASN.1 strings that are used to represent time are defined to always be a subset of ASCII. All ASN.1 string types are encoded with some type of ASCII-subset/derivative. In fact the period above on L489 should be a 0x2E (ASCII `.`) so it's not misinterpreted as EBCDIC 0x4B.